src/components/expanded-state/custom-gas/FeesPanel.js

```diff
@@ -0,0 +1,509 @@
+      const maxPriorityFeePerGas =
+        selectedGasFee?.gasFeeParams?.maxPriorityFeePerGas;
+
+      const gweiMaxPriorityFeePerGas = Number(maxPriorityFeePerGas?.gwei || 0);
+      const gweiMaxFeePerGas = Number(maxFeePerGas?.gwei || 0);
+
+      const newGweiMaxPriorityFeePerGas =
+        Math.round((gweiMaxPriorityFeePerGas + priorityFeePerGas) * 100) / 100;
+      const newGweiMaxFeePerGas =
+        Math.round((gweiMaxFeePerGas + feePerGas) * 100) / 100;
+
+      const newMaxPriorityFeePerGas = parseGasFeeParam(
+        gweiToWei(newGweiMaxPriorityFeePerGas)
+      );
+      const newMaxFeePerGas = parseGasFeeParam(gweiToWei(newGweiMaxFeePerGas));
+
+      if (newMaxPriorityFeePerGas.amount < 0 || newMaxFeePerGas.amount < 0)
+        return;
+
+      setCustomMaxPriorityFee(get(newMaxPriorityFeePerGas, 'gwei', 0));
+      setCustomMaxBaseFee(parseInt(get(newMaxFeePerGas, 'gwei', 10)));
+
+      const newGasParams = {
+        ...selectedGasFee.gasFeeParams,
+        maxFeePerGas: newMaxFeePerGas,
+        maxPriorityFeePerGas: newMaxPriorityFeePerGas,
+      };
+      updateToCustomGasFee(newGasParams);
+    },
+    [selectedGasFee.gasFeeParams, updateToCustomGasFee]
+  );
+
+  const addMinerTip = useCallback(() => {
+    updateGasFee({
+      priorityFeePerGas: calculateMinerTipAddDifference(maxPriorityFee),
+    });
+  }, [updateGasFee, maxPriorityFee]);
+
+  const substMinerTip = useCallback(() => {
+    updateGasFee({
+      priorityFeePerGas: -calculateMinerTipSubstDifference(maxPriorityFee),
+    });
+  }, [updateGasFee, maxPriorityFee]);
+
+  const addMaxFee = useCallback(() => {
+    updateGasFee({ feePerGas: GAS_FEE_INCREMENT });
+  }, [updateGasFee]);
+
+  const substMaxFee = useCallback(
+    () => updateGasFee({ feePerGas: -GAS_FEE_INCREMENT }),
+    [updateGasFee]
+  );
+
+  const onMaxBaseFeeChange = useCallback(
+    ({ nativeEvent: { text } }) => {
+      text = text || 0;
+      const maxFeePerGas = parseGasFeeParam(gweiToWei(text));
+      const gweiMaxFeePerGas = maxFeePerGas?.gwei || 0;
+
+      const newGweiMaxFeePerGas = Math.round(gweiMaxFeePerGas * 100) / 100;
+
+      const newMaxFeePerGas = parseGasFeeParam(gweiToWei(newGweiMaxFeePerGas));
+
+      if (newMaxFeePerGas.amount < 0) return;
+
+      setCustomMaxBaseFee(text);
+
+      const newGasParams = {
+        ...selectedGasFee.gasFeeParams,
+        maxFeePerGas: newMaxFeePerGas,
+      };
+      updateToCustomGasFee(newGasParams);
+    },
+    [selectedGasFee.gasFeeParams, updateToCustomGasFee]
+  );
+
+  const onMinerTipChange = useCallback(
+    ({ nativeEvent: { text } }) => {
+      text = text || 0;
+      const maxPriorityFeePerGas = parseGasFeeParam(gweiToWei(text));
+      const gweiMaxPriorityFeePerGas = maxPriorityFeePerGas?.gwei || 0;
+
+      const newGweiMaxPriorityFeePerGas =
+        Math.round(gweiMaxPriorityFeePerGas * 100) / 100;
+
+      const newMaxPriorityFeePerGas = parseGasFeeParam(
```

Comment:
same here: not sure why you need to round and re-parse this

Reply:
similar as above but keep the text as rounded

Reply:
https://github.com/rainbow-me/rainbow/pull/2422/commits/817a2698e2beff1f769315b251a97fa5f94fa061